### DISCUSSION TITLE
Suggested documentation for asp.net openapi configuration

### DISCUSSION THREAD

**Discussion by mrhockeymonkey** [scalar/scalar#4330] [Ideas]
After watching dotnetconf I wanted to give this a go in my asp.net project and I have managed to get it working but it took me a while to figure out how to get OAuth2 + PKCE working correctly. More specifically it took me a while to find out I needed `x-usePkce` extension which was added in #3732 

For a long while I thought this was something that needed to be configured in scalar options but it turns out its in the openapi side of things and I needed to add this snippet to enable it 

```cs
builder.Services.AddOpenApi(options =>
{
    options.AddSecurityScheme(AuthConstants.OAuth2, scheme =>
    {
        scheme.Type = SecuritySchemeType.OAuth2;
        scheme.Flows = new OpenApiOAuthFlows()
        {
            AuthorizationCode = new OpenApiOAuthFlow()
            {
                AuthorizationUrl = new Uri("https://identityserver/connect/authorize"),
                TokenUrl = new Uri("https://identityserver/connect/token"),
                Scopes = new Dictionary<string, string>()
                {
                    ["playground-api"] = "Playground API"
                },
                Extensions = new Dictionary<string, IOpenApiExtension>()
                {
                    ["x-usePkce"] = new OpenApiString("plain")
                }
            }
        };
    });
});
```

It could be a good idea to include some docs on openapi setup for asp.net even if it is technically a different library all-together? If you agree I would happily give it a shot and raise a PR.

**Comment by xC0dex** [comment]
Hi @mrhockeymonkey,

I agree that there isn't much documentation available for the new `Microsoft.AspNetCore.OpenApi` and `ScalarAspNetCore` packages. Personally, I wouldn't suggest adding this to the documentation of the `Scalar.AspNetCore` package, as it’s indeed a different type and focus of documentation.

Let's stay in touch and build something awesome in 2025!

Regarding the `Pkce`: I _think_ it's also planned to make this configuration possible in the Scalar options.
